### ISSUE TITLE
`FermionicOp.set_label_display_mode` affects construction of operator

### ISSUE THREAD

**Issue by jlapeyre** [Qiskit/qiskit-nature#233]
IIUC,  `FermionicOp.set_label_display_mode` should only affect printing. But, in fact it seems to affect interpreting labels in some way.

This works fine:
```python
from qiskit_nature.drivers import PySCFDriver, UnitsType
from qiskit_nature.problems.second_quantization.electronic.builders.fermionic_op_builder import build_ferm_op_from_ints
from qiskit_nature.operators.second_quantization import FermionicOp

def get_molecule():
    h2 = 'H .0 .0 0.7414; H .0 .0 .0'
    basis0 = "sto3g"

    driver = PySCFDriver(atom=h2,
                         unit=UnitsType.ANGSTROM,
                         charge=0,
                         spin=0,
                         basis=basis0)
    molecule = driver.run()
    return molecule

def get_qkn_op(molecule):
    one_body = molecule.one_body_integrals
    two_body = molecule.two_body_integrals
    fop_qkn = build_ferm_op_from_ints(one_body, two_body)
    return fop_qkn

FermionicOp.set_label_display_mode('dense')

molecule = get_molecule()
fop = get_qkn_op(molecule)
```

However, if I change the string `'dense'` to `'sparse'`, then the script raises an error like this:
```python
~/code/github/Qiskit/oneenv/qiskit-nature/qiskit_nature/problems/second_quantization/electronic/builders/fermionic_op_builder.py in _build_ferm_op_helper(one_body_integrals, two_body_integrals)
     85     #  initial labels prevents from an empty labels list when building a FermionicOp)
     86     base_ops_labels.append(initial_label_with_ceoff)
---> 87     fermionic_op = FermionicOp(base_ops_labels)
     88 
     89     return fermionic_op

~/code/github/Qiskit/oneenv/qiskit-nature/qiskit_nature/operators/second_quantization/fermionic_op.py in __init__(self, data, register_length)
    188         ]
    189         if invalid_labels:
--> 190             raise ValueError(f"Invalid labels for sparse labels are given: {invalid_labels}")
    191 
    192         self._register_length = (

ValueError: Invalid labels for sparse labels are given: ['IIII']
```

#### Versions

qiskit_nature: a0f4c7bb3
qiskit-terra: 0df56b345b

**Comment by mrossinek** [comment]
@ikkoham Could you please have a look at this?

**Comment by ikkoham** [comment]
@jlapeyre This issue was fixed by #263. Could you check it and if confirmed, please close this issue?

**Comment by jlapeyre** [comment]
Yes, #263 has fixed the issue.
